content/200-orm/050-overview/500-databases/950-cloudflare-d1.mdx

```diff
@@ -0,0 +1,47 @@
+
+Prisma ORM support for Cloudflare D1 is currently in [Preview](/orm/more/releases#preview). We would appreciate your feedback [on GitHub](tbd).
+
+</TopBlock>
+
+## What is Cloudflare D1?
+
+D1 is Cloudflare’s native serverless database. It’s based on SQLite and can be used when deploying applications with Cloudflare. D1 was initially [launched in 2022](https://blog.cloudflare.com/introducing-d1) and is currently in a public beta stage.
+
+Following Cloudflare’s principles of geographic distribution and bringing compute and data closer to application users, D1 supports automatic read-replication. It dynamically manages the number of database instances and locations of read-only replicas based on how many queries a database is getting, and from where.
+
+For write-operations, queries travel to a single primary instance in order to propagate the changes to all read-replicas and ensure data consistency.
+
+## Commonalities with other database providers
+
+D1 is based on SQLite.
+
+Many aspects of using Prisma ORM with D1 are just like using Prisma ORM with any other relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/orm/prisma-schema)
+- use Prisma ORM's existing [`sqlite` database connector](/orm/overview/databases/sqlite) in your schema
+- use [Prisma Client](/orm/prisma-client) in your application to talk to the database server at D1
+
+## Differences to consider
+
+There are a number of differences between D1 and SQLite to consider. You should be aware of the following when deciding to use D1 and Prisma ORM:
+
+- **Local and remote D1 (SQLite) databases**. Cloudflare provides local and remote versions of D1. The [local](https://developers.cloudflare.com/d1/configuration/local-development) version is managed using the `--local` option of the `wrangler d1` CLI and is located in `.wranger/state`. The [remote](https://developers.cloudflare.com/d1/configuration/remote-development) version is managed by Cloudflare and is accessed via HTTP.
+- **Making schema changes**. Since D1 uses HTTP to connect to the remote database, this makes it incompatible with Prisma Migrate. However, you can use [`prisma migrate diff`](/orm/reference/prisma-cli-reference#migrate-diff) to create a schema migration and then apply the changes to your database using the [`wrangler d1 execute`](https://developers.cloudflare.com/workers/wrangler/commands/#execute) command of the [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/).
+
+## How to connect to D1 in Cloudflare Workers or Cloudflare pages
+
+When using Prisma ORM with D1, you need to use the `sqlite` database provider and the `@prisma/adapter-d1` [driver adapter](/orm/overview/databases/database-drivers#driver-adapters).
```

Comment:
Would it be worth including where to configure this? Or is it not necessary because it's in the instructions you linked after anyways?

Reply:
To me it seems that this indeed chooses to fully leave that detail in the other location.

Reply:
I would also leave this as is unless someone feels strongly about adding more context here?